packages/mui-joy/src/styles/styleFunctionSx.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  Interpolation,
+  unstable_createStyleFunctionSx,
+  compose,
+  style,
+  display,
+  flexbox,
+  grid,
+  positions,
+  sizing,
+  spacing,
+  border,
+  borderTop,
+  borderRight,
+  borderBottom,
+  borderLeft,
+  typographyVariant,
+} from '@mui/system';
+import { JoyTheme, SxProps } from './defaultTheme';
+
+const borderColor = style({
+  prop: 'borderColor',
+  themeKey: 'palette',
+  enableThemeVars: true,
+});
+
+const borderTopColor = style({
+  prop: 'borderTopColor',
+  themeKey: 'palette',
+  enableThemeVars: true,
+});
+
+const borderRightColor = style({
+  prop: 'borderRightColor',
+  themeKey: 'palette',
+  enableThemeVars: true,
+});
+
+const borderBottomColor = style({
+  prop: 'borderBottomColor',
+  themeKey: 'palette',
+  enableThemeVars: true,
+});
+
+const borderLeftColor = style({
+  prop: 'borderLeftColor',
+  themeKey: 'palette',
+  enableThemeVars: true,
+});
+
+const borderRadius = style({
+  prop: 'borderRadius',
+  themeKey: 'radius',
+  enableThemeVars: true,
+});
+
+const borders = compose(
+  border,
+  borderTop,
+  borderRight,
+  borderBottom,
+  borderLeft,
+  borderColor,
+  borderTopColor,
+  borderRightColor,
+  borderBottomColor,
+  borderLeftColor,
+  borderRadius,
+);
+
+const color = style({
+  prop: 'color',
+  themeKey: 'palette',
+  enableThemeVars: true,
+});
+
+const bgcolor = style({
+  prop: 'bgcolor',
+  cssProperty: 'backgroundColor',
+  themeKey: 'palette',
+  enableThemeVars: true,
+});
+
+const backgroundColor = style({
+  prop: 'backgroundColor',
+  themeKey: 'palette',
+  enableThemeVars: true,
+});
+
+const palette = compose(color, bgcolor, backgroundColor);
+
+const boxShadow = style({
+  prop: 'boxShadow',
+  themeKey: 'shadow',
+  enableThemeVars: true,
+});
+
+export const fontFamily = style({
+  prop: 'fontFamily',
+  themeKey: 'fontFamily',
+  enableThemeVars: true,
+});
+
+export const fontSize = style({
+  prop: 'fontSize',
+  themeKey: 'fontSize',
+  enableThemeVars: true,
+});
+
+export const fontWeight = style({
+  prop: 'fontWeight',
+  themeKey: 'fontWeight',
+  enableThemeVars: true,
+});
+
+export const letterSpacing = style({
+  prop: 'letterSpacing',
+  themeKey: 'letterSpacing',
+  enableThemeVars: true,
+});
+
+export const lineHeight = style({
+  prop: 'lineHeight',
+  themeKey: 'lineHeight',
+  enableThemeVars: true,
```

Comment:
I would even add a different key, something that will hld the value `vars`, maybe something like:
```
designTokensKey: 'vars'
```

If it is not defined, the design tokens will be considered first level of the theme.

Reply:
This sounds better.

Reply:
What do you think about `themeNode`?

```js
themeNode: 'vars', // means theme.vars will be used to get `themeKey`
```

Reply:
`themeNode` inside a `theme` sounds strange, but yeah, let's keep it open in case we found a better name

Reply:
Yeah, this does not need to be a final name since only Joy is using this. In the future, this field might not be needed at all if `theme.vars` is official for using CSS variables.

Reply:
For unblocking the PR I would name it `internal_designTokensKey`. We can find a better name later

Reply:
Okay, works for me.